modules/transactions/src/main/java/org/apache/ignite/internal/tx/TxStateMetaFinishing.java

```diff
@@ -56,6 +62,23 @@ public CompletableFuture<TransactionMeta> txFinishFuture() {
         throw new UnsupportedOperationException("Can't get commit timestamp from FINISHING transaction state meta.");
     }
 
+    @Override
+    public TxStateMetaFinishingMessage toTransactionMetaMessage(
+            ReplicaMessagesFactory replicaMessagesFactory,
+            TxMessagesFactory txMessagesFactory
+    ) {
+        TablePartitionId commitPartitionId = commitPartitionId();
+
+        return txMessagesFactory.txStateMetaFinishingMessage()
+                .txStateInt(txState().ordinal())
+                .txCoordinatorId(txCoordinatorId())
+                .commitPartitionId(commitPartitionId == null ? null : toTablePartitionIdMessage(replicaMessagesFactory, commitPartitionId))
```

Comment:
Builder messages don't get @Nullable annotations, let's add a ticket to fix that?

Reply:
Yeah why not.
https://issues.apache.org/jira/browse/IGNITE-22651
https://issues.apache.org/jira/browse/IGNITE-22652